mobile/native/src/ln_dlc/mod.rs

```diff
@@ -156,16 +162,40 @@ pub fn get_seed_phrase() -> Vec<String> {
         .get_seed_phrase()
 }
 
+/// Gets the seed from the storage or from disk. However it will panic if the seed can not be found.
+/// No new seed will be created.
+fn get_seed() -> Bip39Seed {
+    match SEED.try_get() {
+        Some(seed) => seed.clone(),
+        None => {
+            let seed_dir = config::get_seed_dir();
+
+            let network = config::get_network();
+            let seed_path = Path::new(&seed_dir).join(network.to_string()).join("seed");
+            assert!(seed_path.exists());
+
+            let seed = Bip39Seed::initialize(&seed_path).expect("to read seed file");
+            SEED.set(seed.clone());
+            seed
+        }
+    }
+}
+
 pub fn get_node_key() -> SecretKey {
-    NODE.get().inner.node_key()
+    let seed = get_seed();
+    let time_since_unix_epoch = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .expect("unix epos to not be earlier than now");
+    let keys_manger = KeysManager::new(
+        &seed.lightning_seed(),
+        time_since_unix_epoch.as_secs(),
+        time_since_unix_epoch.subsec_nanos(),
```

Comment:
This logic changed. Before we used the `KeysManager` from the node, now you re-initiate it. The `KeysManager` is not deterministic though (see the system time. 
Does this logic have any side effects? 
Why did you change the logic?

Reply:
> Why did you change the logic?

I changed it because I need the secret key for getting the public key and decrypting a backup before starting the node.

Reply:
> Does this logic have any side effects?

Judging from the api docs, there shouldn't be any side effects unless the `starting_time` is unique, every time we are calling that function.

> starting_time isn't strictly required to actually be a time, but it must absolutely, without a doubt, be unique to this instance. ie if you start multiple times with the same seed, starting_time must be unique to each run. Thus, the easiest way to achieve this is to simply use the current time (with very high precision).
The seed MUST be backed up safely prior to use so that the keys can be re-created, however, obviously, starting_time should be unique every time you reload the library - it is only used to generate new ephemeral key data (which will be stored by the individual channel if necessary).

Reply:
However, I adapted the implementation so that it will always return the secret key from the node, if the node is set.

Reply:
b7820829136c01832bd12a06e1aa14e35bcd23b1